### PR TITLE
Containerd v1.0.0 alpha2

### PR DESCRIPTION
**Pull request by ijc**
**- What I did**

Bumped containerd to [v1.0.0-alpha2](https://github.com/containerd/containerd/releases/tag/v1.0.0-alpha2). https://github.com/containerd/containerd/pull/1141 has been merged so this is now tracking mainline containerd again.

Repaired the `linuxkit.packages.containerd` test which was marked skipped in #2298. /cc @AkihiroSuda 

Note that I previously suggested that I would roll up runc rc4 with this update, but that hasn't actually happened yet so that is not done.

**- How I did it**

containerd bump itself was pretty trivial, change `tools/alpine/Dockerfile` to define the new commit and the `pkg/containerd/Dockerfile` `FROM`. Some minor build fixes for `pkg/containerd/cmd/` (see commit message).

reenabling the test required enabling networking in the test container (so dhcp + `/etc/resolv.conf` bind), binding in the binaries (`contianerd*` and `runc`), adding `/etc/localtime` and giving a writable `cgroups` mount.

**- How to verify it**

Run any image, or the `linuxkit.packages.containerd` test.

**- Description for the changelog**
Now uses containerd v1.0.0-alpha2.

**- A picture of a cute animal (not mandatory but encouraged)**
[:format(jpeg):mode_rgb():quality(40)/discogs-images/A-465937-1381785412-1174.jpeg.jpg "Iron Monkey, Iron Monkey")](https://en.wikipedia.org/wiki/Iron_Monkey_(band))

### CODE REVIEW

File: test/cases/040_packages/003_containerd/test-containerd.yml [hunk 1]
```diff
@@ -4,16 +4,21 @@ kernel:
...
+    image: linuxkit/dhcpcd:17423c1ccced74e3c005fd80486e8177841fe02b
+    command: ["/sbin/dhcpcd", "--nobackground", "-f", "/dhcpcd.conf", "-1"]
   - name: sysctl
     image: linuxkit/sysctl:d1a43c7c91e92374766f962dc8534cf9508756b0
   - name: test
-    image: linuxkit/test-containerd:0e350ce958b379b9ddb2bcdd22aaf4508f6ddbc3
+    image: linuxkit/test-containerd:b662189ccd7be5f943b6597e4e9824fd91b322e7
+    mounts:
+     - type: cgroup
+       options: ["rw","nosuid","noexec","nodev","relatime"]
```
Comment (justincormack) [inline test/cases/040_packages/003_containerd/test-containerd.yml]: As this package already has labels for most of its options, better to add this there too.

Comment (ijc) [inline test/cases/040_packages/003_containerd/test-containerd.yml]: Done

**Comment by ijc** [comment]
I just ran
```
 LANG=C sort tools/alpine/versions.x86_64  | sponge tools/alpine/versions.x86_64
```
and it undid the change I made here, so I think this might be me...

Shall I append a commit with LANG=C in the right place and the resulting (un)update to the versions file? I don't propose to actually bump the whole chain the the resulting changed hash though.

**Comment by justincormack** [comment]
May as well do that now, yes.

**Comment by ijc** [comment]
[CI said](https://linuxkit.datakit.ci/log/saved/linuxkit-build-of-886870c93512babc718402cec7689f58ddb6d153-default-ci-pr/befe184b1133add5493613ed9796902cdcbce6eb):
```
[+  0m 26s] ssh: connect to host 104.199.92.51 port 22: Connection refused
[...]
[2017-07-27 11:16.51] Failed: Command "tar" "cf" "-" "." failed with signal -7
```
Some sort of glitch I think. Restarting.

**Comment by justincormack** [comment]
Yes I think that happens very occasionally

**Pull request merged**